frontend/www/js/omegaup/components/login/Signup.test.ts

```diff
@@ -0,0 +1,228 @@
+
+    expect(
+      wrapper.find('button[name="sign_up"]').attributes().disabled,
+    ).toBeFalsy();
+
+    //await wrapper.find('button[name="sign_up"]').trigger('click');
+
+    // An warning should be displayed, but there is no status div
+  });
+
+  it('should handle a complete registration for Under13 user', async () => {
+    const wrapper = mount(login_Signup, {
+      propsData: {
+        validateRecaptcha: false,
+      },
+    });
+
+    await wrapper.findComponent(omegaup_DatePicker).setValue('2012-01-01');
+
+    const policyPrivacyCheckbox = wrapper.find('input[type="checkbox"]')
+      .element as HTMLInputElement;
+
+    await policyPrivacyCheckbox.click();
+
+    await wrapper.find('input[name="reg_username"]').setValue('Omegaup');
+    await wrapper
+      .find('input[name="reg_parent_email"]')
+      .setValue('parentEmail@gmail.com');
+    await wrapper.find('input[name="reg_password"]').setValue('pass12345678');
+    await wrapper
+      .find('input[name="reg_password_confirmation"]')
+      .setValue('pass12345678');
+
+    await wrapper.find('button[name="sign_up"]').trigger('click');
+
+    expect(wrapper.emitted('register-and-login')).toEqual([[expectedValues]]);
+  });
+
+  it('should handle an uncomplete registration for Over13 user', async () => {
+    const wrapper = mount(login_Signup, {
+      propsData: {
+        validateRecaptcha: false,
+      },
+    });
+
+    await wrapper.findComponent(omegaup_DatePicker).setValue('2009-01-01');
+
+    // All the fields in the form enabled when birthdate is filled
+    expect(
+      wrapper.find('input[name="reg_username"]').attributes().disabled,
+    ).toBeFalsy();
+    expect(
+      wrapper.find('input[name="reg_email"]').attributes().disabled,
+    ).toBeFalsy();
+    expect(
+      wrapper.find('input[name="reg_password"]').attributes().disabled,
+    ).toBeFalsy();
+    expect(
+      wrapper.find('input[name="reg_password_confirmation"]').attributes()
+        .disabled,
+    ).toBeFalsy();
+    expect(
+      wrapper.find('input[type="checkbox"]').attributes().disabled,
+    ).toBeFalsy();
+    expect(
+      wrapper.find('input[name="reg_birthdate"]').attributes().disabled,
+    ).toBeFalsy();
+
+    // Now, parent email field doesn't exist because user is Over13, so they should provide their own email
+    expect(wrapper.find('input[name="reg_parent_email"]').exists()).toBe(false);
+
+    // The button should be enabled until policy privacy is checked
+    expect(
+      wrapper.find('button[name="sign_up"]').attributes().disabled,
+    ).toBeTruthy();
+
+    const policyPrivacyCheckbox = wrapper.find('input[type="checkbox"]')
+      .element as HTMLInputElement;
+
+    await policyPrivacyCheckbox.click();
+
+    expect(
+      wrapper.find('button[name="sign_up"]').attributes().disabled,
+    ).toBeFalsy();
+
+    //await wrapper.find('button[name="sign_up"]').trigger('click');
+
+    // An warning should be displayed, but there is no status div
```

Comment:
Now, here we cuold test whether a cass `was-validated` is present:
```suggestion
    expect(wrapper.find('form').classes('was-validated')).toBeFalsy();

    await wrapper.find('button[name="sign_up"]').trigger('click');

    expect(wrapper.find('form').classes('was-validated')).toBeTruthy();
```